pkg/providers/gcp/provider_redis.go

```diff
@@ -154,6 +161,10 @@ func (p *RedisProvider) createRedisInstance(ctx context.Context, networkManager
 		return nil, croType.StatusMessage(statusMessage), errorUtil.Wrap(err, statusMessage)
 	}
 	if maintenanceWindowEnabled {
+		//rescudule maintenance to start immediately
+		foundInstance.MaintenanceSchedule = &redispb.MaintenanceSchedule{
+			StartTime: timestamppb.New(time.Now()),
+		}
```

Comment:
I'm wondering if this has been tested at all - looking at the redispb library it states that `MaintenanceSchedule` is output-only -> https://github.com/integr8ly/cloud-resource-operator/blob/mgdapi-3425-gcp/vendor/google.golang.org/genproto/googleapis/cloud/redis/v1/cloud_redis.pb.go#L641-L643

Reply:
Thank you, yes, I saw it, but maybe need try. If will not work, we can use REST, same way as gcloud

Reply:
I have commented out a nil pointer error and tested this and it is not possible to set like this:

```
"error": "failed to update gcp redis instance projects/REDACTED/locations/europe-west2/instances/REDACTED: rpc error: code = InvalidArgument desc = Invalid option 'maintenance_schedule': 'maintenance_schedule' cannot be updated on existing instances.\ncom.google.apps.framework.request.StatusException: <eye3 title='INVALID_ARGUMENT'/> generic::INVALID_ARGUMENT: Invalid option 'maintenance_schedule': 'maintenance_schedule' cannot be updated on existing instances.\nerror details: name = BadRequest field = update_mask.paths.0 desc = Invalid value: maintenance_schedule", "errorVerbose": "rpc error: code = InvalidArgument desc = Invalid option 'maintenance_schedule': 'maintenance_schedule' cannot be updated on existing instances.\ncom.google.apps.framework.request.StatusException: <eye3 title='INVALID_ARGUMENT'/> generic::INVALID_ARGUMENT: Invalid option 'maintenance_schedule': 'maintenance_schedule' cannot be updated on existing instances.\nerror details: name = BadRequest field = update_mask.paths.0 desc = Invalid value: maintenance_schedule\nfailed to update gcp redis instance
```

Reply:
Hi @adam-cattermole , the logic for Redis - updated. Also, as discussed, it's using now `RescheduleMaintenance()` function that was added to CRO redis client.  Thank you for review and suggestions.